### ISSUE TITLE
MaterialIcons change values ..

### ISSUE THREAD

**Issue by YassirMSS** [flutter/flutter#77797]
Hello,

Im faced an issue when lunching my application in release mode.. all MaterialIcons are changed and not showing properly..

for example :

![image](https://user-images.githubusercontent.com/34966818/110624440-815b6b00-819e-11eb-871e-6b1eca3d7408.png)

home icon are shown as motion_photos_on icon

NB: "uses-material-design: true" added on pubspec.yaml

any solutions?

>> Flutter 1.12.13+hotfix.5 • channel stable

**Comment by TahaTesser** [comment]
Hi @YassirMSS 
You seem to be using a very old Flutter version 
Can you please upgrade to the latest  `stable` and try again
`flutter channel stable`
`flutter upgrade --force`
`flutter doctor -v`

If the problem persists, Can you please provide your `flutter doctor -v` with the upgrade and `flutter run --verbose` logs, and a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example)?
Thank you

**Comment by Leadrive** [comment]
Yes, I meet the same bug.
Maybe on the stable 2.0.3, my icons display ok.
But after I upgrade to the latest master, some icons display wrong, not its value.
Not all is wrong.
Such as:
IconData(int.parse("e486", radix: 16), fontFamily: 'MaterialIcons');
e486 is ok, it will display Icons.people.

IconData(int.parse("e47a", radix: 16), fontFamily: 'MaterialIcons');
e47a is not ok, the correct value is Icons.password, but it will display another icon.

**Comment by Leadrive** [comment]
````javascript
Flutter (Channel master, 2.1.0-13.0.pre.365, on Microsoft Windows [Version 10.0.17763.1697], locale zh-CN)
    • Flutter version 2.1.0-13.0.pre.365 at C:\flutter
    • Framework revision f86c859bea (14 hours ago), 2021-04-01 03:49:02 +0530
    • Engine revision 70af54fa0b
    • Dart version 2.13.0 (build 2.13.0-180.0.dev)
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn
````

**Comment by guidezpl** [comment]
Can you please provide the output of `flutter run -v ...` as well as the contents of the file `C:\flutter\bin\cache\artifacts\material_fonts.stamp`?

**Comment by Leadrive** [comment]
@guidezpl Hi, I don't know how to get the output log what you want.
I delete the c:\flutter\bin\cache and flutter doctor -v
to download the flutter again, the problem still exist.

**Comment by guidezpl** [comment]
If you can run your app from the command line (such as Powershell), can you add the `-v` flag and provide the output? Also, you should be able to get the contents of that file with `type C:\flutter\bin\cache\artifacts\material_fonts.stamp` (alternatively, open it and copy its contents here) Thanks!

**Comment by Leadrive** [comment]
@guidezpl log file is https://www.leadrive.cn/a.txt

**Comment by Leadrive** [comment]
@guidezpl 
C:\flutter\bin\cache\material_fonts.stamp file content is
flutter_infra_release/flutter/fonts/bd151aa3c2f7231344411a01dba4ef61b3cd56b2/fonts.zip

can not search fonts.zip in c:\flutter and its sub dirs.

**Comment by Leadrive** [comment]
@guidezpl 
C:\flutter\bin\cache\material_fonts.stamp file content is
flutter_infra_release/flutter/fonts/bd151aa3c2f7231344411a01dba4ef61b3cd56b2/fonts.zip

can not search fonts.zip in c:\flutter and its sub dirs.

**Comment by Leadrive** [comment]
@guidezpl 
C:\flutter\bin\cache\material_fonts.stamp file content is
flutter_infra_release/flutter/fonts/bd151aa3c2f7231344411a01dba4ef61b3cd56b2/fonts.zip

can not search fonts.zip in c:\flutter and its sub dirs.

**Comment by Leadrive** [comment]
````dart
  Widget build(BuildContext context) {
    return Scaffold(
        appBar: AppBar(
          // Here we take the value from the MyHomePage object that was created by
          // the App.build method, and use it to set our appbar title.
          title: Text(widget.title),
        ),
        body: Center(
            // Center is a layout widget. It takes a single child and positions it
            // in the middle of the parent.
            child: Container(
                padding: EdgeInsets.fromLTRB(20, 0, 20, 20),
                child: TextFormField(
                  initialValue: "",
                  style: TextStyle(color: Colors.black),
                  decoration: InputDecoration(
                    border: OutlineInputBorder(),
                    labelText: 'mobile',
                    icon: Icon(
                      IconData(int.parse("e47a", radix: 16), fontFamily: 'MaterialIcons'),
                      color: Colors.blue,
                    ),
                  ),
                ))));
}
````

**Comment by guidezpl** [comment]
You should use `Icons.password` or `Icons.people` directly rather than refer to the codepoints (e.g. "e47a") which are subject to change every time the font gets updated.

**Comment by Leadrive** [comment]
@guidezpl I want to show the icon from the string, this is a dynamic.
I think this is a bug.

**Comment by guidezpl** [comment]
I'm sorry, I don't understand. Could you rephrase what you're trying to do?

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
